ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/serialisation/ValueSerializer.scala

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.sandbox.stores.ledger.sql.serialisation
+
+import com.digitalasset.daml.lf.value.Value.{AbsoluteContractId, VersionedValue}
+import com.digitalasset.daml.lf.value.ValueCoder.DecodeError
+import com.digitalasset.daml.lf.value.{ValueCoder, ValueOuterClass}
+
+trait ValueSerializer {
+  def serialiseValue(
+      value: VersionedValue[AbsoluteContractId]): Either[ValueCoder.EncodeError, Array[Byte]]
+
+  def deserialiseValue(blob: Array[Byte]): Either[DecodeError, VersionedValue[AbsoluteContractId]]
+}
+
+/**
+  * This is a preliminary serializer using protobuf as a payload type. Our goal on the long run is to use JSON as a payload.
+  */
+object ValueSerializer extends ValueSerializer {
```

Comment:
hashing is quite tricky here, because protobuf serialization is not canonical, that is, there is no single bytestring that corresponds to a protobuf message.

we can merge with the current semantics for now, but please abstract out the hashing in this object (say `valueHasher`) and add a comment noting this pitfall, and linking an issue about this in the SQL milestone.